yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -60,6 +62,9 @@ private[yarn] class YarnAllocator(
 
   import YarnAllocator._
 
+  // RackResolver logs an INFO message whenever it resolves a rack, which is way too often.
+  Logger.getLogger(classOf[RackResolver]).setLevel(Level.WARN)
```

Comment:
Well, I disagree. A user will get very frustrated if they are debugging an issue and they can't turn on the logging. Can you add a check: Logger.getLogger(classOf[RackResolver]).getLevel() != null at least you won't be overriding the logging level if it is set.